paddle/fluid/operators/unsqueeze_op.cc

```diff
@@ -0,0 +1,191 @@
+/* Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#include <string>
+#include <vector>
+#include "paddle/fluid/framework/op_registry.h"
+
+namespace paddle {
+namespace operators {
+
+class UnsqueezeOpInferShape : public framework::InferShapeBase {
+ public:
+  void operator()(framework::InferShapeContext *ctx) const override {
+    PADDLE_ENFORCE(ctx->HasInput("X"),
+                   "Input(X) of UnsqueezeOp should not be null.");
+    PADDLE_ENFORCE(ctx->HasOutput("Out"),
+                   "Output(Out) of UnsqueezeOp should not be null.");
+
+    const auto &axes = ctx->Attrs().Get<std::vector<int>>("axes");
+    PADDLE_ENFORCE(!axes.empty(),
+                   "The unsqueeze axes information must be set by Attr(axes).");
+
+    const auto &x_dims = ctx->GetInputDim("X");
+    // Validity Check: input tensor dims (<6).
+    PADDLE_ENFORCE(static_cast<int>(x_dims.size()) <= 6,
+                   "Invalid dimensions, dynamic dimensions should within "
+                   "[1, 6] dimensions (Eigen limit).");
+    // Validity Check: the range of unsqueeze aixs.
+    for (int axis : axes) {
+      PADDLE_ENFORCE(axis < 6,
+                     "Invalid dimensions, input axis should within "
+                     "[1, 6] dimensions (Eigen limit).");
+    }
```

Comment:
Attribution's checking can be placed in Make, like this [here](https://github.com/PaddlePaddle/Paddle/blob/3d99e2e9df6333c52bca678c06c3ea51344b7894/paddle/fluid/operators/detection/anchor_generator_op.cc#L103-L114).

Reply:
get, thanks

Reply:
Done